action/protocol/staking/contractstaking/indexer.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package contractstaking
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/iotexproject/iotex-address/address"
+
+	"github.com/iotexproject/iotex-core/db"
+)
+
+const (
+	// StakingContractAddress  is the address of system staking contract
+	// TODO (iip-13): replace with the real system staking contract address
+	StakingContractAddress = "io19ys8f4uhwms6lq6ulexr5fwht9gsjes8mvuugd"
+
+	maxBlockNumber uint64 = math.MaxUint64
+)
+
+type (
+	// Indexer is the contract staking indexer
+	// Main functions:
+	// 		1. handle contract staking contract events when new block comes to generate index data
+	// 		2. provide query interface for contract staking index data
+	// Generate index data flow:
```

Comment:
flow explanation should not be here

Reply:
removed and should be added in #3863 at `PutBlock`